### PR TITLE
Initial commit to check release pipeline for github pypi artifacts.

### PR DESCRIPTION
**Pull request by averma-canonical**

### CODE REVIEW

**Review by deusebio — COMMENTED** [review]
Great work @averma-canonical !!!! 

Very neat, clean and modular workflows. There are a few points where I believe we need @taurus-forever input, but the PR looks quite in well shape

File: .github/workflows/upstream-updates-check.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          path: ./SPARK_VERSION
+          write-mode: overwrite
+          contents: |
+            ${{ needs.check-spark.outputs.version }}
+
+      - name: create pull request
+        id: cpr
+        uses: peter-evans/create-pull-request@v4
+        with:
+          token: ${{ secrets.SPARK_PKG_PUBLISH }}
```
Comment (taurus-forever) [inline .github/workflows/upstream-updates-check.yaml:70]: Sharing secrets with 3rd party is a bad idea.
See. https://github.com/canonical/mysql-k8s-operator/pull/131#pullrequestreview-1247858206
Prepared the specification: https://docs.google.com/document/d/1OTV6l8LY0uBe4LehpP-Nd3QWB-2qXdxUJ3DYqka3w04/edit

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: Verify PR and handoff to reviewer
+
+on:
+  push:
```
[the remark this answers is not in the document]
Comment (averma-canonical) [reply .github/workflows/pr.yaml:4]: Hi @deusebio , PR can also evolve with more commits as comments come in.  I thought the 'on.pull_request' directive will not re-run the tests on new commits to the PR branch.

**Review by taurus-forever — COMMENTED** [review]
Can you please use the common naming for workflows, e.g.
https://github.com/canonical/mysql-bundle/tree/main/.github/workflows
or 
https://github.com/canonical/zookeeper-rock/tree/3.6/stable/.github/workflows

Just to keep some common logic between repos.
In general the direction is correct, but we need to clean one topic mentioned in a comment. Will share a decision later today.

File: .github/workflows/pr.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (averma-canonical) [reply .github/workflows/pr.yaml:4]: Please ignore, type 'synchronize' can handle the case I mentioned above.

on:
  pull_request:
    types: [ assigned, opened, synchronize, reopened ]

**Review by deusebio — APPROVED** [review]
Looks good! Once merged we should probably re-open a PR on the main branch canonical/spark-client-snap

**Comment by averma-canonical** [comment]
Merging this within this fork. As Enrico mentioned a new proper PR with be created in canonical repo to sync these changes back.

**Pull request merged**